### PR TITLE
feat: support fragments offline

### PR DESCRIPTION
**Pull request by mchandak29**
### why?
This PR introduces the support of [Fragments](https://www.hlx.live/developer/block-collection/fragment) on offline channels.

### what has changed?
While generating the manifest,  we add assets coming from all Fragments present in the page.

Prerequisite: This requires a helix indexing with "fragments" column been setup in franklin repo (see [example](https://github.com/hlxscreens/cec-screens/blob/53936b128c470682844d9651829739ae3fe75ef0/helix-query.yaml#L68))

### CODE REVIEW

File: src/createManifest.js [hunk 1]
```diff
@@ -122,13 +128,36 @@ export default class ManifestGenerator {
...
+      allEntries.set(entry.path, entry);
+    });
+    // add entries for all fragments
+    const fragmentsList = JSON.parse(fragments);
+
+    // eslint-disable-next-line no-restricted-syntax
+    for (const fragmentPath of fragmentsList) {
+      // eslint-disable-next-line no-unused-vars
+      const [{ entries: newEntries }, _] = await ManifestGenerator
+        .createManifest(host, manifestMap, fragmentPath, false, [`${fragmentPath}.plain.html`]);
```
Comment (habansal) [inline src/createManifest.js:144]: Why is this needed?
Will the fragment plain html be cached as well?
Are we caching the channel plain html as well? if yes, where?

Comment (habansal) [reply src/createManifest.js:144]: as discussed with @mchandak29, Fragment block code makes a call to fragment plain html.
For channel, that is not required

**Review by habansal — APPROVED** [review]

File: src/createManifest.js [hunk 2]
```diff
@@ -28,7 +30,7 @@ export default class ManifestGenerator {
   /**
    * Checks if the image is hosted in franklin
    */
-  static isMedia = (path) => path.trim().startsWith(Constants.MEDIA_PREFIX);
+  static isMedia = (path) => path.trim().includes(Constants.MEDIA_PREFIX);
```
Comment (absarasw) [inline src/createManifest.js:33]: Is there any reason for this change?

File: src/createManifest.js [hunk 3]
```diff
@@ -122,13 +128,36 @@ export default class ManifestGenerator {
...
+        // rebase media URLs to current path
+        if (ManifestGenerator.isMedia(entry.path)) {
+          entry.path = ManifestGenerator.extractMediaFromPath(entry.path);
+          entry.path = PathUtils.getParentFromPath(path).concat(entry.path);
+        }
+        allEntries.set(entry.path, entry);
+      });
+    }
+
+    // bug??
```
Comment (absarasw) [inline src/createManifest.js:155]: bug?

File: src/createManifest.js (hunk 1 above)
Comment (absarasw) [reply src/createManifest.js:144]: @mchandak29 , @habansal 
`For channel, that is not required`
Do you mean for channels we don't need  plaint.html of the channel to be included in manifest?

**Comment by absarasw** [comment]
@mchandak29 , have we used fragments in any of our channel/doc yet and tested the module?

File: src/createManifest.js (hunk 1 above)
Comment (habansal) [reply src/createManifest.js:144]: yes, correct

**Review by habansal — APPROVED** [review]
